### PR TITLE
Fix karma test and travis ci

### PR DESCRIPTION
**Pull request by DKbyo**
**Trello**: https://trello.com/c/DOp7OxIv/111-pass-tests-on-calculator-ui

**Fix**: Fix karma test in components(only render) with firefox, include the airbnb eslint fixes in #34 

**Smoke test**: Don't need it, only change tests

### CODE REVIEW

**Comment by DKbyo** [comment]
@Arbolista I need administrator privileges in order to connect with travis.

File: karma.conf.js [hunk 1]
```diff
@@ -2,24 +2,44 @@
 
 const webpack_config = require('./client/config/webpack/test');
 
+
+const PREPROCESSORS = ['babel', 'webpack', 'sourcemap'];
+
 module.exports = function (config) {
   config.set({
-    browsers: ['PhantomJS'],
+    browsers: ['Firefox'],
```
Comment (manue1) [inline karma.conf.js]: Do we want to run the tests in Chrome too? I saw that you added its launcher.

File: karma.conf.js [hunk 2]
```diff
@@ -2,24 +2,44 @@
...
+    basePath: './',
+    babelPreprocessor: {
+      options: {
+        presets: ['es2015', 'react'],
+      },
+    },
     files: [
-      'test.client.js',
+      'node_modules/babel-polyfill/dist/polyfill.js',
+      'client/**/*.test.js',
```
Comment (manue1) [inline karma.conf.js]: Currently we don't have tests in client/*

File: shared/components/pages/take_action/action/action.test.js [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
-import React from 'react';
+import wrapWithContext from '../../../application/application.component.mock';
 
 import Action from './action.component';
 
 describe('Action component', () => {
   it('renders without problems', (done) => {
-    const action = TestUtils.renderIntoDocument(React.createElement(Action));
-    expect(action.state).toEqual({});
+    const action = TestUtils.renderIntoDocument(wrapWithContext(Action));
```
Comment (manue1) [inline shared/components/pages/take_action/action/action.test.js]: Currently throws an error warning for not having specified a required prop. Easy fix with
```js
const action = TestUtils.renderIntoDocument(
      wrapWithContext(Action, {}, { is_assumption: false }));
```

File: shared/components/pages/take_action/external_offset/external_offset.test.js [hunk 4]
```diff
@@ -1,14 +1,29 @@
...
+          cta: {
+            show_equation: true,
+            title: 'Mock',
+            carbon_price_per_ton: 1,
+            description: '',
+            button_title: '',
+            offset_url: '',
+          },
+        },
+      }),
```
Comment (manue1) [inline shared/components/pages/take_action/external_offset/external_offset.test.js:23]: Another missing required prop warning -> add:
```json
totalUserFootprint: '48.473001',
```

**Review by manue1 — COMMENTED** [review]
Looks good, just a few warnings that we can still get rid of.

File: shared/components/pages/take_action/external_offset/external_offset.test.js (hunk 4 above)
Comment (DKbyo) [reply shared/components/pages/take_action/external_offset/external_offset.test.js:23]: Done!

File: karma.conf.js (hunk 2 above)
Comment (DKbyo) [reply karma.conf.js]: That's right... I remove this line..

File: shared/components/pages/take_action/action/action.test.js (hunk 3 above)
Comment (DKbyo) [reply shared/components/pages/take_action/action/action.test.js]: Done!

File: karma.conf.js (hunk 1 above)
Comment (DKbyo) [reply karma.conf.js]: I had problems with Chrome... but now works ok... I've enable Chrome again

**Pull request merged**